pkg/util/ec2/network.go

```diff
@@ -18,7 +18,7 @@ import (
 var publicIPv4Fetcher = cachedfetch.Fetcher{
 	Name: "EC2 Public IPv4 Address",
 	Attempt: func(ctx context.Context) (interface{}, error) {
-		return getMetadataItem(ctx, imdsIPv4, false)
+		return getMetadataItem(ctx, imdsIPv4, UseIMDSv2(false, false))
```

Comment:
We'll need some renaming likely. Having `UseIMDSv2()` and `useIMDSv2` in the same file but being completely different seems like a developer foot-gun.

Reply:
I renamed the enum it''s fixed